File: 037-两个链表的第一个公共结点/037-FindFirstCommonNode.cpp
/*
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};*/
class Solution {
public:
    ListNode* FindFirstCommonNode( ListNode* pHead1, ListNode* pHead2) {
        if (pHead1 ==nullptr || pHead2==nullptr){
            return NULL;
        }
        ListNode* ap=pHead1;
        ListNode* bp=pHead2;
        while(ap!=bp){
            ap= ap==nullptr?pHead2:ap->next;
            bp= bp==nullptr?pHead1:bp->next;
        }
        return ap;
    }
};

File: 037-两个链表的第一个公共结点/037-FindFirstCommonNode.py
# -*- coding:utf-8 -*-
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None
class Solution:
    def FindFirstCommonNode(self, headA, headB):
        # write code here
        if headA is None or headB is None:
            return None
        aprior=headA
        bprior=headB
        while aprior != bprior:
            aprior=headB if aprior is None else aprior.next
            bprior=headA if bprior is None else bprior.next
        return aprior
